### PR TITLE
Add annotations and py.typed to conform with PEP561

### PR DESCRIPTION
**Pull request by keiclone**
*Issue number of the reported bug or feature request: #24

**Describe your changes**
made `attrs-strict` PEP561 compatible by adding annotations and `py.typed`. This removes the missing import error from `mypy` when attempting to run static type checking on packages using `attrs-strict`

**Testing performed**
Was able to successfully run static type checking in a test project after installing attrs-strict from my branch.

when testing on a separate library:
```python
# test.py
@attr.s
class FooClass(object):
    list_of_numbers = attr.ib(validator=type_validator(), type=List[int])


fc = FooClass([1, 2, 3, "four"])
```

```bash
# before changes
$ pipenv install attrs-strict
Installing git+https://github.com/bloomberg/attrs-strict
...
$ mypy test.py
error: Cannot find implementation or library stub for module named 'attrs_strict'
note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
error: List item 3 has incompatible type "str"; expected "int"

# after changes
$ pipenv install git+https://github.com/keiclone/attrs-strict#egg=attrs-strict
Installing git+https://github.com/keiclone/attrs-strict#egg=attrs-strict
...
$ mypy test.py
error: List item 3 has incompatible type "str"; expected "int"
```

**Additional context**
could use some extra attention on how this was added to tox to make sure it was done correctly

future changes to `attrs_strict` will need to ensure the stubs are updated appropriately as this will not come for free as long as we need to support python 2

### CODE REVIEW

**Comment by erikseulean** [comment]
This is great, thanks for the hard work. Could you please squash your commits ? 

@gaborbernat since you have the most experience on this, I'd like to hear your comments on this.

**Comment by erikseulean** [comment]
Could you please capitalise the commit and add a description of what it does ? Helps with the docs...

**Comment by keiclone** [comment]
@erikseulean made a few changes. lemme know if they work?

File: setup.cfg [hunk 1]
```diff
@@ -1,2 +1,12 @@
 [bdist_wheel]
 universal = true
+
+[mypy]
+python_version = 3.7
```
Comment (gaborbernat) [inline setup.cfg:5]: This should be not set here as we need to run against both python 2 and 3, given we support both.

File: setup.cfg [hunk 2]
```diff
@@ -1,2 +1,12 @@
...
+namespace_packages=True
```
Comment (gaborbernat) [inline setup.cfg:6]: We're not using namespace packages so I don't think we need this. Can we also not cherry-pick individual configs but just make it globally strict?

File: attrs_strict/__init__.pyi [hunk 3]
```diff
@@ -0,0 +1,2 @@
+from ._error import AttributeTypeError as AttributeTypeError, BadTypeError as BadTypeError, TupleError as TupleError, TypeValidationError as TypeValidationError, UnionError as UnionError
```
Comment (gaborbernat) [inline attrs_strict/__init__.pyi:1]: we need some tool that validates that content from the python file matches with the stubs; for example https://pypi.org/project/retype/ is an example, please use it in our tox check...  see https://github.com/ambv/retype/blob/master/tox.ini#L58

File: setup.py [hunk 4]
```diff
@@ -13,6 +14,12 @@
     ]
 )
 
+package_data = {"attrs_strict": ["py.typed"]}
+for root, dirs, files in os.walk("attrs_strict"):
```
Comment (gaborbernat) [inline setup.py:18]: A better approach to add these via a recursive include https://docs.python.org/3/distutils/sourcedist.html

File: setup.cfg (hunk 1 above)
Comment (keiclone) [reply setup.cfg:5]: can remove this. we'd have to set two different runs in tox then, one for each version

File: setup.cfg (hunk 2 above)
Comment (keiclone) [reply setup.cfg:6]: can remove this as well. I'm not aware of a way to make it globally strict in [mypy](https://mypy.readthedocs.io/en/stable/config_file.html)

File: attrs_strict/__init__.pyi (hunk 3 above)
Comment (keiclone) [reply attrs_strict/__init__.pyi:1]: wasn't aware of this. will add!

File: setup.cfg (hunk 2 above)
Comment (gaborbernat) [reply setup.cfg:6]: Just pass in the --strict flag in tox.ini invocation?

File: setup.cfg (hunk 1 above)
Comment (gaborbernat) [reply setup.cfg:5]: yeah, can do that in tox.ini as separate commands one after another

**Comment by erikseulean** [comment]
This will need rebase.

**Pull request closed without merging**